src/NServiceBus.Transport.SQS/InputQueuePump.cs

```diff
@@ -405,13 +459,21 @@ async Task DeleteMessage(Message message, string s3BodyKey, CancellationToken me
             try
             {
                 // should not be canceled
-                await sqsClient.DeleteMessageAsync(inputQueueUrl, message.ReceiptHandle, messageProcessingCancellationToken).ConfigureAwait(false);
+                await sqsClient
+                    .DeleteMessageAsync(inputQueueUrl, message.ReceiptHandle, messageProcessingCancellationToken)
+                    .ConfigureAwait(false);
             }
             catch (ReceiptHandleIsInvalidException ex)
             {
                 Logger.Info($"Message receipt handle {message.ReceiptHandle} no longer valid.", ex);
                 return; // if another receiver fetches the data from S3
             }
+            catch (Exception ex) when (!ex.IsCausedBy(messageProcessingCancellationToken))
+            {
+                Logger.Info($"Failed to delete message with receipt handle {message.MessageId}");
+            }
+
+            messagesToBeDeleted.AddOrUpdate(message.MessageId, true);
```

Comment:
We only need to do this if `DeleteMessageAsync` fails, so it should be inside a `catch`